doc/api/util.md

```diff
@@ -2060,6 +2074,151 @@ console.log(values, positionals);
 // Prints: [Object: null prototype] { foo: true, bar: 'b' } []
 ```
 
+### `parseArgs` help text
+
+`parseArgs` supports automatic formatted help text generation for command-line options. When
+general help text is provided, a help option is automatically added unless disabled or already present.
+
+#### Simple usage
+
+By default, providing general help text automatically adds a help option (`-h, --help`) and
+returns formatted help text in `result.helpText`:
+
+```mjs
+import { parseArgs } from 'node:util';
+
+const options = {
+  foo: {
+    type: 'boolean',
+    short: 'f',
+    help: 'use the foo filter',
+  },
+  bar: {
+    type: 'string',
+    help: 'use the specified bar filter',
+  },
+};
+
+const result = parseArgs({
+  help: 'utility to control filters',
+  options,
+});
+
+if (result.values.help) {
+  console.log(result.helpText);
+  // Prints:
+  // utility to control filters
+  // -f, --foo                     use the foo filter
+  // --bar <arg>                   use the specified bar filter
+  // -h, --help                    Show help
+}
+```
+
+#### Custom help option
+
+You can override the auto-added help option by defining your own:
+
+```mjs
+import { parseArgs } from 'node:util';
+
+const options = {
+  foo: {
+    type: 'boolean',
+    short: 'f',
+    help: 'use the foo filter',
+  },
+  bar: {
+    type: 'string',
+    help: 'use the specified bar filter',
+  },
+  help: {
+    type: 'boolean',
+    short: '?',
+    help: 'display help',
+  },
+};
+
+const result = parseArgs({
+  help: 'utility to control filters',
+  options,
+});
+
+if (result.values.help) {
+  console.log(result.helpText);
+  // Prints:
+  // utility to control filters
+  // -f, --foo                     use the foo filter
+  // --bar <arg>                   use the specified bar filter
+  // -?, --help                    display help
+}
+```
+
+#### Advanced configuration
+
+Use `addHelpOption` and `returnHelpText` to customize help behavior:
+
+```mjs
+import { parseArgs } from 'node:util';
+
+const options = {
+  foo: {
+    type: 'boolean',
+    short: 'f',
+    help: 'use the foo filter',
+  },
+  assist: {
+    type: 'boolean',
+    short: '?',
+    help: 'display help',
+  },
+};
+
+const result = parseArgs({
+  addHelpOption: false, // Suppress auto help option
```

Comment:
why is this needed? since no text was provided, the default should be `false` here - and enabling `returnHelpText` (which, per above, i'm not even sure ever should be `false` or have an option) doesn't automatically make it default to `true`.